apisix/plugins/redirect.lua

```diff
@@ -129,17 +159,34 @@ function _M.rewrite(conf, ctx)
         end
     end
 
-    if uri and ret_code then
-        local new_uri, err = concat_new_uri(uri, ctx)
-        if not new_uri then
-            core.log.error("failed to generate new uri by: ", uri, " error: ",
-                           err)
-            return 500
+    if ret_code then
+        if uri then
+            local new_uri, err = concat_new_uri(uri, ctx)
+            if not new_uri then
+                local msg = "failed to generate new uri by: " .. uri .. err
+                core.log.error(msg)
```

Comment:
I think it's better that way.
```suggestion
                core.log.error("failed to generate new uri by: " .. uri .. err)
```